x-pack/plugins/ml/public/application/services/ml_api_service/data_frame_analytics.ts

```diff
@@ -32,6 +33,17 @@ interface GetDataFrameAnalyticsResponse {
   data_frame_analytics: DataFrameAnalyticsConfig[];
 }
 
+export interface DeleteDataFrameAnalyticsWithIndexStatus {
+  success: boolean;
+  error?: CustomHttpResponseOptions<ResponseError>;
+}
+interface DeleteDataFrameAnalyticsWithIndexResponse {
```

Comment:
I guess it'd be better to define somewhere in `x-pack/plugins/ml/common/types` because we need this interface both server and client-side